ray_data_eval/libsolver/src/solver/environment.rs

```diff
@@ -562,6 +613,35 @@ impl Environment {
         false
     }
 
+    fn collect_all_events(&self) -> Vec<TraceEvent> {
+        let mut all_events = Vec::new();
+        for executor in self.executors.iter() {
+            all_events.extend(executor.timeline_json.clone());
+        }
+        all_events
+    }
+
+    fn write_all_events_to_json(&self) {
+        let all_events = self.collect_all_events();
+        let json = match serde_json::to_string(&all_events) {
```

Comment:
A lazy way of handling errors is just to write `let json = serde_json::to_string(&all_events)!;` (the exclamation mark).

Reply:
I see! I think u meant the question mark (from I saw in the docs). Btw the easiest thing I could do here is just to append a `.unwrap()`, which returns either the inner element or `panic`. Does this also work?

```
    fn write_all_events_to_json(&self) {
        let all_events = self.collect_events_from_all_executors();
        let json = serde_json::to_string(&all_events).unwrap();
        let mut file = File::create("output.json").unwrap();
        file.write_all(json.as_bytes()).unwrap();
    }
```